src/frequenz/sdk/timeseries/_formula_engine/_formula_printer.py

```diff
@@ -0,0 +1,220 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+from __future__ import annotations
+import enum
+
+from ._formula_steps import (
+    Adder,
+    Averager,
+    Clipper,
+    ConstantValue,
+    Divider,
+    FormulaStep,
+    MetricFetcher,
+    Multiplier,
+    OpenParen,
+    Subtractor,
+)
+
+class OperatorPrecedence(enum.Enum):
+    """Represents the precedence of an operator."""
+
+    Subtraction = 1
+    Addition = 2
+    Division = 3
+    Multiplication = 4
+    Primary = 9
+
+    def __gt__(self, other: OperatorPrecedence) -> bool:
+        """Return whether the precedence of this operator is greater than the precedence of the other operator.
+
+        Args:
+            other: The other operator (on the right-hand side).
+
+        Returns:
+            bool: Whether the precedence of this operator is greater than the precedence of the other operator.
```

Comment:
Please don't specify types in the docs, the tooling is able to get it from the code anyways.